pypika/tests/test_selects.py

```diff
@@ -263,15 +264,26 @@ def test_groupby__str(self):
         self.assertEqual('SELECT "foo",COUNT(DISTINCT *) FROM "abc" GROUP BY "foo"', str(q))
 
     def test_groupby__alias(self):
-        q = Query.from_(self.t).select(
-            fn.Sum(self.t.foo).as_('bar'),
-        ).groupby(
-            fn.Sum(self.t.foo).as_('bar'),
-        )
-
-        self.assertEqual('SELECT SUM("foo") "bar" FROM "abc" GROUP BY SUM("foo")', str(q))
-
-    def test_groupby_with_case_does_not_show_an_alias(self):
+        bar = self.t.bar.as_('bar01')
+        q = Query.from_(self.t) \
+            .select(fn.Sum(self.t.foo), bar) \
+            .groupby(bar)
+
+        self.assertEqual('SELECT SUM("foo"),"bar" "bar01" FROM "abc" GROUP BY "bar01"', str(q))
```

Comment:
is there also a test where a column has an alias but the query is grouped by the column name (not the alias)? if not, might be worth adding, I think.

Reply:
yes, most of the group by tests

Reply:
hmm, I now looked through all of them with aliases in selects and we always use the alias in the group by clause as well.
what I meant is having an alias in a select and then using the original column name in the group by, e.g. (I think, I'm still not very familiar with the syntax):
```
bar = self.t.bar.as_('bar01')
          q = Query.from_(self.t) \
              .select(fn.Sum(self.t.foo), bar) \
              .groupby(self.t.bar)
```

Reply:
In your example it will group by `SUM("foo")` because it uses the alias to determine if the alias can be used in the group by.

Reply:
right!, probably without the SUM in the select is also fine. either way, I think it's a good case to add